File: 15-Java8/src/by/gsu/epamlab/beans/LightTrial.java
package by.gsu.epamlab.beans;

public class LightTrial extends Trial{
    private static final int FIRST_EXAM_PASS_MARK = 40;
    private static final int SECOND_EXAM_PASS_MARK = 60;

    public LightTrial() {
    }

    public LightTrial(String name, int mark1, int mark2) {
        super(name, mark1, mark2);
    }

    @Override
    public Trial getCopy(){
        return new LightTrial(getName(), getMark1(), getMark2());
    }

    @Override
    public boolean isPassed() {
        return getMark1() >= FIRST_EXAM_PASS_MARK && getMark2() >= SECOND_EXAM_PASS_MARK;
    }
}

File: 22-jsOop/beans/LightTrial.js
const Trial = require("./Trial.js");

class LightTrial extends Trial{
    static FIRST_EXAM_PASS_MARK = 40;
    static SECOND_EXAM_PASS_MARK = 60;

    getCopy(){
        return new LightTrial(this._account, this._mark1, this._mark2);
    }

    isPassed() {
        return this._mark1 >= LightTrial.FIRST_EXAM_PASS_MARK && this._mark2 >= LightTrial.SECOND_EXAM_PASS_MARK;
    }

}
module.exports = LightTrial;